### PR TITLE
Fix #4176: Added dark mode support to profile activities and app version activity

### LINKED ISSUE #4176: Add dark mode support to ProfileEditActivity, ProfileListActivity, ProfileRenameActivity, ProfileResetPinActivity and AppVersionActivity.
Add dark mode support to ProfileEditActivity, ProfileListActivity, ProfileRenameActivity, ProfileResetPinActivity and AppVersionActivity.

Go through this [wiki](https://github.com/oppia/oppia-android/wiki/Dark-Mode) to get started.

#### Location in the app
1. Open the app
2. Go to Administrator Controls
3. Go to edit profile
4. These activities can be found navigated to through there.

### PR DESCRIPTION
**Pull request by aayushimathur6**
## Explanation
Fixes #4176

Profile List: https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/ddc3dadf-10e4-45f6-aeea-eb8a330e00ff/specs/

Profile Edit: https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/693301de-5907-4f6a-b183-037412478267/

Profile Rename: https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/d0227921-1c24-4bb7-8667-a8fa5f4742dd/

Profile Reset: https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/32c9276d-072c-4edf-abf9-435323bb475a/

App Version: https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/79e0846d-de5c-42b7-93aa-f93d4995616d/

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing
- 
| BEFORE  | AFTER |
| ------------- | ------------- |
| Screenshot_20220916_153310  | Screenshot_20220916_152556  |
| Screenshot_20220902_164755  | Screenshot_20220902_165035  |
| Screenshot_20220916_153741  | Screenshot_20220916_152840  |
| Screenshot_20220916_153924  | Screenshot_20220916_152923  |
| Screenshot_20220916_154020  | Screenshot_20220916_153037  |

### CODE REVIEW

**Comment by aayushimathur6** [comment]
@rt4914 made a new pull request for issue no. 4176 (https://github.com/oppia/oppia-android/issues/4176), PTAL thanks

File: app/src/main/res/drawable/general_item_background_border_grey.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+  android:shape="rectangle">
+  <solid android:color="@color/component_color_profile_rename_activity_secondary_background_color" />
+  <stroke
+    android:width="1dp"
+    android:color="@color/color_def_black_12" />
+</shape>
```
Comment (rt4914) [inline app/src/main/res/drawable/general_item_background_border_grey.xml]: EOF is missing, add one blank line at the end.

Comment (rt4914) [reply app/src/main/res/drawable/general_item_background_border_grey.xml]: Also, please use a component color for stroke too.

File: app/src/main/res/drawable/general_item_background_border_grey.xml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+  android:shape="rectangle">
+  <solid android:color="@color/component_color_profile_rename_activity_secondary_background_color" />
```
Comment (rt4914) [inline app/src/main/res/drawable/general_item_background_border_grey.xml]: Don't use `profile, activity, background` like terms in naming for this file.

`component_color_general_item_background_solid_color`

File: app/src/main/res/layout/profile_edit_fragment.xml [hunk 3]
```diff
@@ -142,7 +143,7 @@
         android:id="@+id/profile_edit_allow_download_container"
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        android:background="@color/color_def_white"
+        android:background="@color/component_color_profile_edit_activity_profile_edit_allow_download_container"
```
Comment (rt4914) [inline app/src/main/res/layout/profile_edit_fragment.xml]: `component_color_profile_edit_activity_profile_edit_allow_download_container_color`

All component colors should end with suffix `_color`

File: app/src/main/res/values-night/color_palette.xml [hunk 4]
```diff
@@ -88,4 +88,10 @@
   <color name="color_palette_drag_drop_single_item_gradient_start_color">@color/color_def_accessible_grey</color>
   <color name="color_palette_drag_drop_single_item_stroke_color">@color/color_def_white</color>
   <color name="color_palette_forward_arrow_button_color">@color/color_def_oppia_turquoise</color>
+  <color name="color_palette_profile_view_designation_color">@color/color_def_oppia_turquoise</color>
+  <color name="color_palette_primary_toolbar_color">@color/color_def_forest_green</color>
+  <color name="color_palette_secondary_text_color">@color/color_def_oppia_silver</color>
+
+
+
```
Comment (rt4914) [inline app/src/main/res/values-night/color_palette.xml]: Remove these extra blank lines

**Review by rt4914 — CHANGES_REQUESTED** [review]
@aayushimathur6 Suggested changes.

File: app/src/main/res/drawable/general_item_background_border_grey.xml (hunk 1 above)
Comment (aayushimathur6) [reply app/src/main/res/drawable/general_item_background_border_grey.xml]: done

File: app/src/main/res/drawable/general_item_background_border_grey.xml (hunk 2 above)
Comment (aayushimathur6) [reply app/src/main/res/drawable/general_item_background_border_grey.xml]: done

File: app/src/main/res/layout/profile_edit_fragment.xml (hunk 3 above)
Comment (aayushimathur6) [reply app/src/main/res/layout/profile_edit_fragment.xml]: done

File: app/src/main/res/values-night/color_palette.xml (hunk 4 above)
Comment (aayushimathur6) [reply app/src/main/res/values-night/color_palette.xml]: done

**Comment by aayushimathur6** [comment]
@rt4914 PTAL

**Comment by aayushimathur6** [comment]
@rt4914  also for the one check failing then do I need to add any regex pattern?

File: app/src/main/res/drawable/general_item_background_border_grey.xml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+  android:shape="rectangle">
+  <solid android:color="@color/component_color_general_item_background_solid_color" />
+  <stroke
+    android:width="1dp"
+    android:color="@color/component_color_general_item_secondary_background_stroke_color" />
+
+</shape>
```
Comment (rt4914) [inline app/src/main/res/drawable/general_item_background_border_grey.xml]: EOF is missing, add one blank line at the end.

File: app/src/main/res/drawable/general_item_background_border_grey.xml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+  android:shape="rectangle">
+  <solid android:color="@color/component_color_general_item_background_solid_color" />
+  <stroke
+    android:width="1dp"
+    android:color="@color/component_color_general_item_secondary_background_stroke_color" />
+
```
Comment (rt4914) [inline app/src/main/res/drawable/general_item_background_border_grey.xml]: Remove this blank line

**Review by rt4914 — CHANGES_REQUESTED** [review]
Screenshot 2022-09-15 at 1 05 16 AM


In this and similar `deactivated-save-button` the border color does not match with that of mocks. It should be dull https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/32c9276d-072c-4edf-abf9-435323bb475a/specs/

Also, have a look at this to understand how to use `8-characters` color code correctly 
https://github.com/oppia/oppia-android/wiki/Working-on-UI#color-code

File: app/src/main/res/drawable/general_item_background_border_grey.xml (hunk 5 above)
Comment (aayushimathur6) [reply app/src/main/res/drawable/general_item_background_border_grey.xml]: Applied, i was confused about the blank line but now I have corrected PTAL, Thanks

File: app/src/main/res/drawable/general_item_background_border_grey.xml (hunk 6 above)
Comment (aayushimathur6) [reply app/src/main/res/drawable/general_item_background_border_grey.xml]: done

**Comment by aayushimathur6** [comment]
@rt4914 corrected the color, PTAL

**Comment by aayushimathur6** [comment]
> Screenshot 2022-09-15 at 1 05 16 AM
> 
> In this and similar `deactivated-save-button` the border color does not match with that of mocks. It should be dull https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/32c9276d-072c-4edf-abf9-435323bb475a/specs/
> 
> Also, have a look at this to understand how to use `8-characters` color code correctly https://github.com/oppia/oppia-android/wiki/Working-on-UI#color-code

done

**Comment by aayushimathur6** [comment]
@rt4914 updated the content cell as well with the lightening boundary color of the save button according to the android working on UI wiki

File: app/src/main/res/values/component_colors.xml [hunk 7]
```diff
@@ -256,5 +257,33 @@
...
+  <color name="component_color_profile_edit_activity_profile_edit_allow_download_heading_text_color">@color/color_palette_highlighted_text_color</color>
+  <!-- Profile List Activity -->
+  <color name="component_color_profile_list_activity_background_color">@color/color_palette_primary_background_color</color>
+  <color name="component_color_profile_list_activity_profile_name_text_color">@color/color_palette_dark_text_color</color>
+  <color name="component_color_profile_list_activity_profile_view_designation_text_color">@color/color_palette_profile_view_designation_color</color>
+  <color name="component_color_general_item_background_border_solid_color">@color/color_palette_container_background_color</color>
+  <!-- Profile Rename Activity -->
+  <color name="component_color_profile_rename_activity_background_color">@color/color_palette_dark_background_color</color>
+  <color name="component_color_start_button_transparent_background_stroke_color">@color/color_palette_save_button_border_color</color>
+  <!-- Profile Reset Pin Fragment -->
```
Comment (rt4914) [inline app/src/main/res/values/component_colors.xml]: `Profile Reset Pin Activity`

**Review by rt4914 — DISMISSED** [review]
@aayushimathur6 Merged with latest develop.

Also, I do think that the border of disabled button is quite bright as compared to mocks. Can you please check why is that, maybe expermient with colors a bit, like trying
`#3BD1C466` and `#663BD1C4` and seeing the difference.

File: app/src/main/res/values/component_colors.xml (hunk 7 above)
Comment (aayushimathur6) [reply app/src/main/res/values/component_colors.xml]: done

**Comment by aayushimathur6** [comment]
> @aayushimathur6 Merged with latest develop.
> 
> Also, I do think that the border of disabled button is quite bright as compared to mocks. Can you please check why is that, maybe expermient with colors a bit, like trying `#3BD1C466` and `#663BD1C4` and seeing the difference.

Done with merging
for this @rt4914 earlier I was applying 
image
this which is too bright i.e #3BD1C4
but after you shared the mock I applied the light version i.e #663BD1C4 but this is named here previously as bright
image
still, the check is failing 
Also, for #3BD1C466 this is not the color that I have to apply you can see the following
image
I am not getting the exact color

**Comment by aayushimathur6** [comment]
> @aayushimathur6 Merged with latest develop.
> 
> Also, I do think that the border of disabled button is quite bright as compared to mocks. Can you please check why is that, maybe expermient with colors a bit, like trying `#3BD1C466` and `#663BD1C4` and seeing the difference.

**Comment by aayushimathur6** [comment]
@rt4914 @BenHenning PTAL

**Comment by BenHenning** [comment]
Sorry, will need to look at this tomorrow.

File: app/src/main/res/values/component_colors.xml [hunk 8]
```diff
@@ -62,6 +62,7 @@
   <color name="component_color_shared_text_input_layout_error_color">@color/color_palette_error_color</color>
   <color name="component_color_general_item_background_solid_color">@color/color_palette_container_background_color</color>
   <color name="component_color_general_item_background_stroke_color">@color/color_palette_dark_background_color</color>
+  <color name="component_color_general_item_secondary_background_stroke_color">@color/Color_palette_stroke_color</color>
```
Comment (BenHenning) [inline app/src/main/res/values/component_colors.xml]: Typo? Does this actually work with the capitalized ``Color``?

File: app/src/main/res/drawable/general_item_background_border_grey.xml (hunk 1 above)
Comment (BenHenning) [reply app/src/main/res/drawable/general_item_background_border_grey.xml]: This seems to have been resolved.

File: app/src/main/res/values/component_colors.xml (hunk 7 above)
Comment (BenHenning) [reply app/src/main/res/values/component_colors.xml]: This seems to be resolved.

**Review by BenHenning — COMMENTED** [review]
Thanks @aayushimathur6. I didn't have too much to comment on, and I checked past comments from Rajat.

PTAL at my follow-up comment and also make sure that CI checks are passing before sending this back.

File: app/src/main/res/values/component_colors.xml (hunk 8 above)
Comment (aayushimathur6) [reply app/src/main/res/values/component_colors.xml]: Thanks @BenHenning , corrected it

**Comment by aayushimathur6** [comment]
> Thanks @aayushimathur6. I didn't have too much to comment on, and I checked past comments from Rajat.
> 
> PTAL at my follow-up comment and also make sure that CI checks are passing before sending this back.

I have looked at your comment I was making a silly mistake, All checks have passed.
PTAL @BenHenning

**Review by BenHenning — APPROVED** [review]
Thanks @aayushimathur6! Approved.

**Pull request merged**